extensions/amp-signal-collection-frame/0.1/amp-signal-collection-frame.js

```diff
@@ -0,0 +1,149 @@
+/* Copyright 2016 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {CONFIG} from './_amp-signal-collection-frame-config';
+import {
+  createElementWithAttributes,
+  removeChildren,
+} from '../../../src/dom';
+import {Layout} from '../../../src/layout';
+import {listen} from '../../../src/event-helper';
+import {isValidAttr} from '../../../src/sanitizer';
+import {user} from '../../../src/log';
+
+/** @const {string} */
+const TAG = 'amp-signal-collection-frame';
+
+/**
+ * @const {number} interval at which messages about touch events is sent
+ * @visibleForTesting
+ */
+export const MESSAGE_INTERVAL_MS = 100;
+
+export class AmpSignalCollectionFrame extends AMP.BaseElement {
```

Comment:
This will be shared across AdSense and Doubleclick and potentially other networks that could return a creative from either of those networks so having it in A4A abstract element makes the most sense IMO.  Let's discuss offline.